### DISCUSSION TITLE
Poor Performance (.NET Native Toolchain Compilation?)

### DISCUSSION THREAD

**Discussion by jsiemensLatitudegeo** [dotnet/maui#19951] [Q&A] [answered]
We're migrating an app from Xamarin Forms to MAUI. We've found that on Windows the performance is generally worse on MAUI and memory consumption is higher. One thing I was wondering about is that on UWP there was an option to enable .NET Native Toolchain Compilation, which provided a good performance boost. There doesn't seem to be that option available in MAUI / WinUI 3. Am I missing something? Are there any other build settings I can enable to improve performance on a release build?

**Comment by p2pbsh** [comment, marked as answer]
Bad news, no, .NET Native was a UWP specific technology which is essentially obsolete now. Good news, Native AOT is its (superior) replacement and is being aggressively developed across the .NET ecosystem. However, MAUI is waiting upon WinUI support, which in turn is waiting on WinRT support, so wouldn't get your hopes up of seeing it anytime soon.

**Reply by jsiemensLatitudegeo** [reply]
Yeah, that was the conclusion I came to as well. Thanks for confirming.
